.circleci/main-config.yml

```diff
@@ -773,12 +770,16 @@ jobs:
         type: string
         default: *api-spec-path
     steps:
-      - checkout
-      - run:
-          name: Build docs
-          command: |
-            cd << parameters.spec_dir >>
-            ./build.sh documentation
+      - run: echo "Doing build-api-docs-job << pipeline.parameters.api-docs >>"
+      - when:
+          condition: << pipeline.parameters.api-docs >>
+          steps:
+            - checkout
+            - run:
+                name: Build docs
+                command: |
+                  cd << parameters.spec_dir >>
+                  ./build.sh documentation
```

Comment:
Our original configuration was completely NOT running this job if there were no changes in api-docs.
This PR is about combining all the jobs into a single workflow that can be status checked.
So this job HAS to run.
But if there are no api-docs changes the job can skip all the real steps.
CircleCI forces us to at least execute one step, so that is why we have the run: echo